cmd/clusterctl/client/config/reader_viper.go

```diff
@@ -82,3 +114,19 @@ func (v *viperReader) Set(key, value string) {
 func (v *viperReader) UnmarshalKey(key string, rawval interface{}) error {
 	return viper.UnmarshalKey(key, rawval)
 }
+
+// checkDefaultConfig checks the existence of the default config
+func (v *viperReader) checkDefaultConfig() bool {
+	var defaultConfigExists bool
+	for _, path := range v.configPaths {
+		for _, ext := range viper.SupportedExts {
+			f := fmt.Sprintf("%s%s.%s", path, ConfigName, ext)
+			_, err := os.Stat(f)
+			if err == nil {
+				defaultConfigExists = true
```

Comment:
nit: maybe just `return true` here instead, for simplicity since you're inside nested loops (and ditch defaultConfigExists)?